### ISSUE TITLE
Template is broken in various ways

### ISSUE THREAD

**Issue by Ramblurr** [nextjournal/clerk-cljs-demo#4]
Unfortunately this template is broken in various ways, it doesn't work out of the box like the README seems to suggest it would (though I could be misunderstanding).

More importantly however the `nextjournal.clerk.static-app` ns was removed [here](https://github.com/nextjournal/clerk/commit/cb19394b59ec1b8d3bf6a733dc4adcaa729a0667#diff-fc0920ccd961ba1b1e3d01365a5a9ef61efd5bd3e2fc9594f5c813e1da2c62cc) so it won't work with newer versions of clerk.

**Comment by Ramblurr** [comment]
It seems the static stuff isn't necessary for just running the notebook locally. To get it running locally I had to:

1. Update the two clerk deps in deps.edn to `ee0248bbc1054476e781e25b11414fa9899050b9`
2. Copy all of the dependencies from clerk's [`package.json`](https://github.com/nextjournal/clerk/blob/ee0248bbc1054476e781e25b11414fa9899050b9/package.json) to my own
3. Remove `nextjournal.clerk.static-app` from `shadow-cljs.edn` 
4. Use `clojure -A:dev ` to launch the repl

Then I was easily able to load cljs libraries (Quil in my case) in the cljs file and render them, very nice! :tada: 

Unfortunately building the static app is still out of my reach. Running `bb build-static` results in 

`Execution error (StringIndexOutOfBoundsException) at jdk.internal.util.Preconditions$1/apply (Preconditions.java:55).`

<img src="https://github.com/nextjournal/clerk-cljs-demo/assets/14830/f7345189-66c3-4d28-8d30-9ec34074238d" width="200">

**Comment by borkdude** [comment]
Perhaps you can debug what is going wrong in `subs`, it could be an off-by-one error or so :)

**Comment by Ramblurr** [comment]
Yup, just did that. That's what I get for posting to github before troubleshooting myself.

The problem with `subs` was that the repo I was doing this from was `git init`ed but had no commits, so it couldn't find the latest sha. 

Now the static app is building, but fails to load in the browser:

`Uncaught ReferenceError: nextjournal is not defined`

![image](https://github.com/nextjournal/clerk-cljs-demo/assets/14830/e70036b0-b1d5-41fb-9c1d-95da2cb5505b)

I suspect that has to do with the changes in [this commit](https://github.com/nextjournal/clerk/commit/cb19394b59ec1b8d3bf6a733dc4adcaa729a0667#diff-fc0920ccd961ba1b1e3d01365a5a9ef61efd5bd3e2fc9594f5c813e1da2c62cc) that removed `nextjournal.clerk.static-app`

Replacing `nextjournal.clerk.static-app` with `nextjournal.clerk.sci-env` in shadow-cljs.edn didn't sort that out..

**Comment by Ramblurr** [comment]
Huzzah! It's working again.

After generating the template:

1. Update the two clerk deps in deps.edn to `ee0248bbc1054476e781e25b11414fa9899050b9` (which is the latest as of 2024-05-27)
2. Copy all of the dependencies from clerk's [`package.json`](https://github.com/nextjournal/clerk/blob/ee0248bbc1054476e781e25b11414fa9899050b9/package.json) to the generated `package.json`
3. Remove `nextjournal.clerk.static-app`  from `shadow-cljs.edn`'s :entries list
4. Add `nextjournal.clerk.sci-env` and  `nextjournal.clerk.trim-image` to `shadow-cljs.edn`'s :entries list
5. `npm install`
6. Use `bb clerk-watch` and `bb build-static`

Step 2 is really quite unfortunate. @sritchie @zampino is there anyway around adding all the clerk npm deps to the project?

**Comment by sritchie** [comment]
@Ramblurr shadow should be doing that automatically for you, that's a bummer. 

fwiw, this was my latest attempt at a template that tries to smooth over these edges: https://github.com/mentat-collective/clerk-utils/tree/main/resources/clerk_utils/custom

I haven't used the one in this project in a while so I'm not sure if the clerk-utils template will help. But give it a try as another option.
